php/libraries/NDB_Page.class.inc

```diff
@@ -590,6 +590,229 @@ class NDB_Page
         return $this->form->createElement('password', $field, $label, $attr);
     }
 
+     /**
+     * Display the page to a logged in user.
+     */
+    public function displayUserPage(\User $user, array $get, array $post) : string {
```

Comment:
@MounaSafiHarab I know you've been making some changes to main.php lately, can you review this part (which is mostly main.php moved into a function) to make sure none of them were accidentally removed?